lib/Pool.js

```diff
@@ -255,6 +259,13 @@ Pool.prototype.escapeId = function escapeId(value) {
   return mysql.escapeId(value, false);
 };
 
+Pool.prototype.format = function(sql, values) {
+  if (typeof this.config.connectionConfig.queryFormat == "function") {
+    return this.config.connectionConfig.queryFormat.call(this, sql, values, this.config.connectionConfig.timezone);
```

Comment:
I have to think about if it's acceptable that the context of the `queryFormat` call is not going to be a `Connection` object in these cases and if this is going to be an issue.